### PR TITLE
Bug/disable timing log

### PR DESCRIPTION
**Pull request by ezturner**

### CODE REVIEW

**Comment by ezturner** [comment]
Investigate the perfomance with the datetime parsing

**Comment by ezturner** [comment]
Also investigate settings.py load/run time

File: restclients_core/dao.py [hunk 1]
```diff
@@ -221,6 +222,14 @@ def get_setting(self, key, default=None):
         key = "RESTCLIENTS_%s" % key
         return getattr(settings, key, default)
 
+    def get_overridable_setting(self, key, default):
```
Comment (fanglinfang) [inline restclients_core/dao.py]: There is already a get_service_setting

File: .travis.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ before_script:
 - python travis-ci/live_ssl_server.py &
 - python travis-ci/live_ssl_bad_server.py &
 script:
-- pep8 restclients_core
+- pep8 restclients_core --exclude=restclients_core/tests
```
[the remark this answers is not in the document]
Comment (ezturner) [reply .travis.yml]: The settings override is very long: 
https://github.com/uw-it-aca/uw-restclients-core/blob/25b5ea854e96962a171c2bd0655651d2e9fea72b/restclients_core/tests/test_timing_log.py#L10-L13

Also things like regids, string resources, etc. can often fail pep8. I took this from https://github.com/uw-it-aca/uw-restclients-sws/blob/master/.travis.yml

File: restclients_core/dao.py (hunk 1 above)
Comment (ezturner) [reply restclients_core/dao.py]: I may be mistaken, but I believe that get_service_setting does not allow to inherit the default global restclient setting.

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fanglinfang) [reply .travis.yml]: Could you define the values before the class?

File: restclients_core/dao.py (hunk 1 above)
Comment (fanglinfang) [reply restclients_core/dao.py]: How about add default to get_service_setting?

File: restclients_core/dao.py [hunk 3]
```diff
@@ -239,18 +244,39 @@ def _getModule(self, value, default_class, args=[]):
...
+                  from_cache, total_time))
+        logger.info(msg)
+
+    def should_log(self):
+        log_timing = self.get_service_setting("TIMING_LOG_ENABLED", False)
+        logging_rate = float(self.get_service_setting("TIMING_LOG_RATE", 1.0))
+
+        # format is ISO 8601
+        log_start_str = self.get_service_setting("TIMING_START", None)
+        log_end_str = self.get_service_setting("TIMING_END", None)
```
Comment (fanglinfang) [inline restclients_core/dao.py]: would it be better to move the above 4 lines into the init method so that they are set only once?

**Pull request merged**